### DISCUSSION TITLE
Refresh the entire app or screen?

### DISCUSSION THREAD

**Discussion by MuongKimhong** [Textualize/textual#3723] [Q&A]
Is there anyway to force refresh the app or screen?

I want to have a button whenever I click the button I want to refresh the screen, just like in web we can use `window.location.reload()` to reload the current page.

```python
class MyScreen(Screen):
    def compose(self) -> ComposeResult:
        # run som_function()

    def on_button_pressed(self, event: Button.Pressed) -> None:
        # refresh the screen so compose() will be called again
```

I want to run `compose` function again & again whenever the button is clicked.

**Comment by TomJGooding** [comment]
Textual calls compose only once, it's not quite equivalent to HTML. Could you provide more detail what you're trying to update/refresh so we can offer some suggestions?
